core/src/main/java/com/google/errorprone/bugpatterns/ClassInitializationDeadlock.java

```diff
@@ -144,7 +144,7 @@ private void handle(ExpressionTree tree) {
         if (!use.isSubClass(classSymbol, state.getTypes())) {
           return;
         }
-        if (!isStatic(use)) {
+        if (use.isEnclosedBy(classSymbol) && !isStatic(use)) {
           // Nested inner classes implicitly take the enclosing instance as a constructor parameter,
           // and can't be initialized without first initializing their containing class.
```

Comment:
From the comment, it's clear the intention was to only match situations where the subclass is a non-static nested class of the superclass. But it's overly broad; any class that isn't a static inner class (since only inner classes can be marked `static`) will be matched by the current conditional. First checking if the class is enclosed by the defining class makes the conditional true to the comment.